fendermint/app/src/app.rs

```diff
@@ -610,13 +611,25 @@ where
         );
         let txs = request.txs.into_iter().map(|tx| tx.to_vec()).collect();
 
+        let (state_params, block_height) =
+            self.state_params_at_height(request.height.value().into())?;
+        let state = FvmExecState::new(
+            self.state_store_clone(),
+            self.multi_engine.as_ref(),
+            block_height as ChainEpoch,
+            state_params,
+        )
+        .context("error creating new state")?;
+
         let txs = self
             .interpreter
-            .prepare(self.chain_env.clone(), txs)
+            .prepare((self.chain_env.clone(), state), txs)
             .await
             .context("failed to prepare proposal")?;
 
         let txs = txs.into_iter().map(bytes::Bytes::from).collect();
+        // TODO: This seems leaky placed here, should be done in `interpreter`, that's where it's ipc
+        // TODO: aware, here might have filtered more important messages.
         let (txs, size) = take_until_max_size(txs, request.max_tx_bytes.try_into().unwrap());
```

Comment:
For example, if you checkout https://github.com/consensus-shipyard/ipc/blob/f4c67489325ec6aa9ded608b2be9118c89cd55e5/fendermint/vm/interpreter/src/chain.rs#L166
`ipc` critical messages are appended to the end of the list. If there are quite some messages, the `ipc` messages might be filtered out as messages are selected from the start.

Since we are introduce some form of message selection, I was looking at some way to put all the selection logic in one place instead of split into different parts of the codebase. I think passing the block limit into the interpreter works for me, at least we have message selection in one place, easier to debug, onboard and maintain.